### PR TITLE
Add GCPCloudSQLSSLV1 constraint template

### PR DESCRIPTION
**Pull request by zhuzeng**
This contraint template requires GCP CloudSQL instance to have SSL
enabled for incoming traffic.

### CODE REVIEW

File: policies/templates/gcp_cloudsql_ssl_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-cloudsql-ssl-v1
+  annotations:
+    # Example of tying a template to a CIS benchmark
+    benchmark: CIS11_5.03
```
Comment (t12g) [inline policies/templates/gcp_cloudsql_ssl_v1.yaml]: Let's delete this annotation unless this section # is correct...

File: validator/test/fixtures/assets/cloudsql/data.json [hunk 2]
```diff
@@ -0,0 +1,185 @@
+[{
+  "name": "//sqladmin.googleapis.com/my-cloudsql",
```
Comment (t12g) [inline validator/test/fixtures/assets/cloudsql/data.json]: can we delete these "fake" data?

File: validator/test/fixtures/assets/cloudsql/data.json [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  "resource": {
+    "settings": {
+      "ipConfiguration": {
+        "requireSsl": true
+      }
+    }
+  }
+},
+{
+  "name": "//cloudsql.googleapis.com/projects/test1",
```
Comment (t12g) [inline validator/test/fixtures/assets/cloudsql/data.json]: this name doesn't quite look right?

File: policies/templates/gcp_cloudsql_ssl_v1.yaml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+            	"details": metadata,
+
+            }] {
+
+            	asset := input.asset
+
+            	asset.asset_type == "sqladmin.googleapis.com/Instance"
+
+            	asset.resource.settings.ipConfiguration.requireSsl == false
```
Comment (zhuzeng) [inline policies/templates/gcp_cloudsql_ssl_v1.yaml]: My template file actually hardcoded this value... which should be provided by a constraint file.

File: validator/test/fixtures/assets/cloudsql/data.json [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        "settingsVersion": "28",
+        "storageAutoResize": true,
+        "storageAutoResizeLimit": "0",
+        "tier": "db-n1-standard-1"
+      },
+      "state": "RUNNABLE"
+    }
+  },
+{
+  "name": "//cloudsql.googleapis.com/projects/noble-history-87417/instances/galu",
```
Comment (t12g) [inline validator/test/fixtures/assets/cloudsql/data.json]: nit: can we use a more descriptive name to indicate whether this is a "good" or "bad" instance?

Comment (t12g) [reply validator/test/fixtures/assets/cloudsql/data.json]: similarly below...

File: validator/cloudsql_ssl.rego [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	asset := input.asset
+	asset.asset_type == "sqladmin.googleapis.com/Instance"
+	asset.resource.settings.ipConfiguration.requireSsl == false
+
+	message := sprintf("%v does not have SSL enabled for incoming traffic", [asset.name])
```
Comment (t12g) [inline validator/cloudsql_ssl.rego]: nit: it's more like "does not require SSL"; clients can still use SSL if this setting is off

File: policies/templates/gcp_cloudsql_ssl_v1.yaml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-cloudsql-ssl-v1
```
[the remark this answers is not in the document]
Comment (zhuzeng) [reply policies/templates/gcp_cloudsql_ssl_v1.yaml]: Done.

File: policies/templates/gcp_cloudsql_ssl_v1.yaml (hunk 1 above)
Comment (zhuzeng) [reply policies/templates/gcp_cloudsql_ssl_v1.yaml]: Done.

File: validator/test/fixtures/assets/cloudsql/data.json (hunk 2 above)
Comment (zhuzeng) [reply validator/test/fixtures/assets/cloudsql/data.json]: Done.

File: validator/test/fixtures/assets/cloudsql/data.json (hunk 5 above)
Comment (zhuzeng) [reply validator/test/fixtures/assets/cloudsql/data.json]: Done.

**Pull request merged**